app/controllers/file.rb

```diff
@@ -275,6 +281,12 @@ def breadcrumbs
 				files = OpenStruct.new(fn: "files", route: site.route("files"))
 				[site, files] + breadcrumbs
 			end
+
+			def render_or_error
+				yield
+			rescue Tint::File::IncompatibleFrontmatter => e
```

Comment:
sinatra will let you install an exception handler for requests. Instead of wrapping every handler in a rescue block